lib/CustomElementRegistry.js

```diff
@@ -0,0 +1,155 @@
+"use strict";
+var xml = require('./xmlnames');
+
+var forbiddenNames = [
+  'annotation-xml',
+  'color-profile',
+  'font-face',
+  'font-face-src',
+  'font-face-uri',
+  'font-face-format',
+  'font-face-name',
+  'missing-glyph'];
+
+var forbiddenExtendsNames = [
+  'bgsound',
+  'blink',
+  'isindex',
+  'multicol',
+  'nextid',
+  'spacer',
+  'elementnametobeunknownelement'
+];
+
+function isInvalidCustomElementName(name) {
+  return !name || !xml.isValidQName(name) || name.indexOf('-') < 0 || forbiddenNames.indexOf(name) > -1 || name.toLowerCase() !== name;
+}
+
+function CustomElementRegistry(registry, document) {
```

Comment:
Haven't noticed anything regarding this at package.json or in the readme. 
@fgnass
Though it makes sense to have engine key in package json. Semantic increment of version shouldn't hurt anyone since Classes supported from 4th version.